basics_of_sst-core/tutorial/Example00.md

```diff
@@ -0,0 +1,568 @@
+                "Clock element example",                // Description                 
+                COMPONENT_CATEGORY_UNCATEGORIZED        // Component category
+            )
+
+        private:
+            // Member variables for this example.
+            //
+            SST::Output logger_;        // For displaying log messages.
+            uint64_t componentId_;      // SST supplied component id.
+            uint64_t clockTicks_;       // Maximum number of clock ticks.
+            uint64_t clockTickCount_;   // Clock ticks counter.
+
+    };  // Close the class
+}   // Close the namespace
+
+#endif
+```
+
+This code defines the component used in the simulation.  The first few lines
+
+```
+#include <sst/core/component.h>
+#include <sst/core/output.h>
+#include <sst/core/params.h>
+```
+
+specify include files needed from the SST core libraries.  The file `component.h` defines the SST base component from which all SST components are derived.  The file `output.h` contains definitions used for logging.  The file `params.h` contains definitions used to process incoming parameters from the simulation configuration file, which will be described later.
+
+The logging levels defined below the include files are for convenience when writing log messages.
+
+Next the namespace is defined.
+
+```
+namespace Example00
+{
+```
+
+This is not strictly necessary but is useful to prevent name clashes.
+
+Finally the component is defined.
+
+```
+    // This is a very simple component.  It only registers a clock and 
+    // prints log messages as the clock handler is called.
+    //
+    // Remember, all components inherit from SST::Component
+    //
+    class ExampleComponent : public SST::Component
+    {
+        public:
+            // Constructor/Destructor
+            //
+            ExampleComponent(SST::ComponentId_t id, SST::Params &params);
+            ~ExampleComponent() {}
+
+            // Standard SST::Component functions.  These all need to
+            // be implemented in the component, even if they are empty.
+            //
+            void setup(void);
+            void finish(void);
+
+            // Clock handler.  This is the method called from the clock event.
+            //
+            bool clockTick(SST::Cycle_t cycle);
+```
+
+As noted in the comments, all SST components are derived from the base component SST::Component or one of its children.  The constructor and destructor are a standard format and required to interface with the SST core functions.  
+
+The `setup()` and `finish()` methods are also required to interface with the SST core functions.  `setup()` is called after the component constructor but before the simulation starts.  As such, it's a good place to put any component initialization that can't be done in the constructor.  `finish()` is called after the simulation ends and it a good place to perform any necessary cleanup associated with the component.
```

Comment:
There is also the init() and complete() functions which are called before setup() and finish() respectively. They can also be used for the same purposes. This comment is  just to double check that you intended to leave those out and focus on setup() and finish().

Reply:
You probably want an explanation of the init() and complete() phases for completeness.  The simulation lifetime at a high level runs in these steps:

construct graph
partition/distribute graph
construct simulation objects (Components, Links, etc)
init phase
setup
run loop
complete phase
finish

Both the init and complete phases are interactive.  The init()/complete() functions will continue to be called until there are no UntimedEvents sent.  For many Components, these phases are critical to getting things setup correctly.